src/Symfony/Component/Security/Core/Encoder/NativePasswordEncoder.php

```diff
@@ -45,7 +45,7 @@ public function __construct(int $opsLimit = null, int $memLimit = null, int $cos
             throw new \InvalidArgumentException('$cost must be in the range of 4-31.');
         }
 
-        $this->algo = \defined('PASSWORD_ARGON2I') ? max(PASSWORD_DEFAULT, \defined('PASSWORD_ARGON2ID') ? PASSWORD_ARGON2ID : PASSWORD_ARGON2I) : PASSWORD_DEFAULT;
+        $this->algo = \defined('PASSWORD_ARGON2ID') ? PASSWORD_ARGON2ID : (\defined('PASSWORD_ARGON2I') ? PASSWORD_ARGON2I : PASSWORD_BCRYPT);
```

Comment:
your plan initially was to always be at least as good as `PASSWORD_DEFAULT`, even if PHP upgrades it. Is this plan gone ?

Reply:
yes, the plan is gone: in 7.4, `PASSWORD_DEFAULT === null` and other `PASSWORD_*` consts are strings; this makes it too difficult to maintain the plan.